src/lib/Libdb/pgsql/db_common.c

```diff
@@ -0,0 +1,1358 @@
+ *
+ * @return	int
+ * @retval	0	- Success but no rows found
+ * @retval	-1	- Failure
+ * @retval	>0	- Success and number of rows found
+ *
+ */
+int
+pbs_db_search(void *conn, pbs_db_obj_info_t *obj, pbs_db_query_options_t *opts, query_cb_t query_cb)
+{
+	void *st;
+	int ret;
+	int totcount;
+	int refreshed;
+	int rc;
+
+	st = db_initialize_state(conn, query_cb);
+	if (!st)
+		return -1;
+
+	ret = db_fn_arr[obj->pbs_db_obj_type].pbs_db_find_obj(conn, st, obj, opts);
+	if (ret == -1) {
+		/* error in executing the sql */
+		db_destroy_state(st);
+		return -1;
+	}
+	totcount = 0;
+	while ((rc = db_cursor_next(conn, st, obj)) == 0) {
+		query_cb(obj, &refreshed);
+		if (refreshed)
+			totcount++;
+	}
+
+	db_destroy_state(st);
+	return totcount;
+}
+
+/**
+ * @brief
+ *	Get the next row from the cursor. It also is used to get the first row
+ *	from the cursor as well.
+ *
+ * @param[in]	conn - Connected database handle
+ * @param[in]	state - The cursor state handle.
+ * @param[out]	pbs_db_obj_info_t - The pointer to the wrapper object which
+ *		describes the PBS object (job/resv/node etc) that is wrapped
+ *		inside it. The row data is loaded into this parameter.
+ *
+ * @return	Error code
+ * @retval	-1  - Failure
+ * @retval	0  - success
+ * @retval	1  - Success but no more rows
+ *
+ */
+int
+db_cursor_next(void *conn, void *st, pbs_db_obj_info_t *obj)
+{
+	db_query_state_t *state = (db_query_state_t *)st;
+	int ret;
+
+	if (state->row < state->count) {
+		ret = db_fn_arr[obj->pbs_db_obj_type].pbs_db_next_obj(conn, st, obj);
+		state->row++;
+		return ret;
+	}
+	return 1; /* no more rows */
+}
+
+/**
+ * @brief
+ *	Delete an existing object from the database
+ *
+ * @param[in]	conn - Connected database handle
+ * @param[in]	pbs_db_obj_info_t - Wrapper object that describes the object
+ *		(and data) to delete
+ *
+ * @return	int
+ * @retval	-1  - Failure
+ * @retval	0   - success
+ * @retval	1   -  Success but no rows deleted
+ *
+ */
+int
+pbs_db_delete_obj(void *conn, pbs_db_obj_info_t *obj)
+{
+	return (db_fn_arr[obj->pbs_db_obj_type].pbs_db_delete_obj(conn, obj));
+}
+
+/**
+ * @brief
+ *	Load a single existing object from the database
+ *
+ * @param[in]	conn - Connected database handle
+ * @param[in/out]pbs_db_obj_info_t - Wrapper object that describes the object
+ *		(and data) to load. This parameter used to return the data about
+ *		the object loaded
+ *
+ * @return      Error code
+ * @retval       0  - success
+ * @retval	-1  - Failure
+ * @retval	 1 -  Success but no rows loaded
+ *
+ */
+int
+pbs_db_load_obj(void *conn, pbs_db_obj_info_t *obj)
+{
+	return (db_fn_arr[obj->pbs_db_obj_type].pbs_db_load_obj(conn, obj));
+}
+
+/**
+ * @brief
+ *	Initializes all the sqls before they can be used
+ *
+ * @param[in]   conn - Connected database handle
+ *
+ * @return      Error code
+ * @retval       0  - success
+ * @retval	-1  - Failure
+ *
+ *
+ */
+int
+db_prepare_sqls(void *conn)
+{
+	if (pbs_db_prepare_job_sqls(conn) != 0)
```

Comment:
why do these need to have a pbs_ prefix? They seem to be internal functions for postgres only and are not part of the DB API